[24기 김동희] ADD: [Mission 1-6 완료] 메인페이지 댓글 좋아요/취소 기능 구현 완료

dev-bomdong opened the pull request:
## 수정 사항 간략한 한줄 요약
ADD: 메인페이지 댓글 좋아요/취소 기능 구현 완료 [Mission 1-6 완료] 
		 
## 수정 사항들 자세한 내용
    - main page 하트 클릭시 붉은 색으로 변경
    - main page 붉은 하트 클릭시 다시 빈 하트로 변경
    - main page 댓글 삭제 버튼 변경

## 기타 질문 및 특이 사항
- 로그인페이지,메인페이지에서 각 요소의 margin, padding, width값을 pixel 위주로 주었는데, 이럴 경우 다른 해상도인 리눅스,맥OS 환경 하에서 다른 레이아웃으로 보이게 되는 것일지, 이런 현상을 막기위해선 모든 요소의 margin값을 vh, vw, % 등으로 주어야 할지 고민됩니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

joonsikyang commented on students/11th/seunghaoh/style/login.css:
```diff
@@ -0,0 +1,135 @@
+:root {
+  /* color */
+  --white-color: white;
+  --grey-light-color: #fafafa;
+  --grey-medium-color: #dbdbdb;
+  --grey-dark-color: #8e8e8e;
+  --skyblue-color: #b2dffc;
+}
```
Nice!🙌

joonsikyang commented on students/11th/seunghaoh/style/login.css:
```diff
@@ -0,0 +1,135 @@
+  --skyblue-color: #b2dffc;
+}
+
+html,
```
html에 스타일 속성을 지정해줄 필요가 있나요?

seungha-oh replied:
로그인 박스를 중앙에 놓고 싶은데, flex 를 쓰는 방법 말고는 좋은 방법을 찾지 못했습니다. html, body 태그에 height: 100% 를 주고 container 의 display 를 flex 하는 방법으로 로그인 박스를 중앙에 배치하였는데요. 다른 방법 조언 부탁드립니다.

joonsikyang replied:
body 태그에만 부여하면 됩니다. html은 화면에 보이는 요소만 담겨있는게 아니기 때문에 body태그에만 부여해주세요.

seungha-oh replied:
네! 수정하겠습니다. 감사합니다!!

joonsikyang commented on students/11th/seunghaoh/style/login.css:
```diff
@@ -0,0 +1,135 @@
+}
+
+main button {
+  width: 250px;
+  margin: 8px 40px;
+  padding: 5px 9px;
+  border: 0 none;
+  border-radius: 4px;
+  font-size: 13px;
+  background-color: var(--skyblue-color);
+  color: var(--white-color);
+  font-weight: bold;
```
하나의 css 요소에 여러가지 속성값을 지정할 때도 우선순위, 관련도 고려해서 순서 위치시켜주세요. 나름의 convention을 잘 갖춰가면서 코드 작성하는 습관 길러주세요 :)

seungha-oh replied:
반영하여 수정했습니다.

joonsikyang commented on students/11th/seunghaoh/style/main.css:
```diff
@@ -0,0 +1,207 @@
+  --font-large: 16px;
+}
+
+html,
```
마찬가지입니다! html 태그에는 스타일 속성 지정할 필요가 있을까요?

seungha-oh replied:
확인 후 수정하였습니다.

joonsikyang commented on students/11th/seunghaoh/style/main.css:
```diff
@@ -0,0 +1,207 @@
+}
+
+html,
+body {
+  height: 100%;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  flex-direction: column;
+}
```
둘은 합쳐주세요!

seungha-oh replied:
합쳤습니다.

seungha-oh commented on students/11th/seunghaoh/index.html:
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+    <title>repl.it</title>
+    <link href="style.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <script src="script.js"></script>
+  </body>
+</html>
```
삭제하였습니다.

seungha-oh commented on students/11th/seunghaoh/login.html:
```diff
@@ -0,0 +1,41 @@
+  </head>
+  <body>
+    <div class="container">
+      <!-- header -->
```
삭제하였습니다.

seungha-oh commented on students/11th/seunghaoh/main.html:
```diff
@@ -0,0 +1,157 @@
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width" />
+    <title>repl.it</title>
```
수정하였습니다.

seungha-oh commented on students/11th/seunghaoh/main.html:
```diff
@@ -0,0 +1,157 @@
+        <div class="icons">
+          <ul>
+            <li>
+              <img
+                src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/explore.png"
+                alt="Explore icon"
+              />
+            </li>
+            <li>
+              <img
+                src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/heart.png"
+                alt="Like icon"
+              />
+            </li>
+            <li>
+              <img
+                src="https://s3.ap-northeast-2.amazonaws.com/cdn.wecode.co.kr/bearu/profile.png"
+                alt="MyPage icon"
+              />
```
확인 후 수정하였습니다.

seungha-oh commented on students/11th/seunghaoh/login.html:
```diff
@@ -0,0 +1,41 @@
+  <body>
+    <div class="container">
+      <!-- header -->
+      <header>
+        <img alt="instagram logo image" src="img/logo_text.png" />
+      </header>
+      <!-- main -->
+      <main>
+        <input
+          type="inputText"
+          placeholder="전화번호, 사용자 이름 또는 이메일"
+        />
+        <input type="password" placeholder="비밀번호" />
+        <button>로그인</button>
+      </main>
+      <!-- footer -->
+      <footer>
+        <div class="askLoginOption">
+          <div class="line"></div>
+          <div class="askText">또는</div>
+          <div class="line"></div>
+        </div>
+        <button>
+          <span class="facebookLogo"></span>
+          <span class="facebookLoginText">Facebook으로 로그인</span>
+        </button>
+        <div class="askForgotPassword">비밀번호를 잊으셨나요?</div>
+      </footer>
```
from 과 div 사용하여 수정했습니다.

joonsikyang commented on students/11th/seunghaoh/js/login.js:
```diff
@@ -0,0 +1,24 @@
+const pw = document.getElementsByClassName("inputPw")[0];
+const thisIsBtn = document.getElementsByClassName("loginBtn")[0];
+
+id.addEventListener("keyup", function () {
+  loginBtnActive();
+});
+
+pw.addEventListener("keyup", function () {
+  loginBtnActive();
+});
```
id와 pw를 감싸는 요소에 이벤트를 부여해보시기 바랍니다. 함수로 하나로 줄일 수 있겠네요!

saengmotmi commented on students/11th/seunghaoh/js/login.js:
```diff
@@ -0,0 +1,40 @@
+// const id = document.getElementsByClassName("inputId")[0];
+// const pw = document.getElementsByClassName("inputPw")[0];
```
주석은 깔끔하게 제거해주세요

saengmotmi commented on students/11th/seunghaoh/js/login.js:
```diff
@@ -0,0 +1,40 @@
+//   loginBtnActive();
+// });
+
+inputs[0].addEventListener("keyup", function () {
+  loginBtnActive();
+});
+
+inputs[1].addEventListener("keyup", function () {
+  loginBtnActive();
+});
```
1. 반복문 사용해서 두 작업을 하나로 합쳐줄 수 있을 것 같습니다.
2. 더 나은 방법은 inputs를 감싸는 상위 요소에 keyup 이벤트 리스너를 달아주는 것입니다. 그러면 이벤트 리스너 하나만으로도 같은 동작을 할 수 있습니다.

saengmotmi commented on students/11th/seunghaoh/js/login.js:
```diff
@@ -0,0 +1,40 @@
+});
+
+function loginBtnActive() {
+  // const idVal = id.value;
+  // const pwVal = pw.value;
+  // if (idVal && pwVal) {
+  //   thisIsBtn.setAttribute("style", "opacity: 1");
+  //   thisIsBtn.disabled = false;
+  // } else {
+  //   thisIsBtn.setAttribute("style", "opacity: 0.4");
+  //   thisIsBtn.disabled = true;
+  // // }
```
주석 정리해주세요

saengmotmi commented on students/11th/seunghaoh/js/login.js:
```diff
@@ -0,0 +1,40 @@
+  //   thisIsBtn.disabled = true;
+  // // }
+  // const inputsVal = inputs.value;
+  if (inputs[0].value && inputs[1].value) {
+    thisIsBtn.setAttribute("style", "opacity: 1");
+    thisIsBtn.disabled = false;
+  } else {
+    thisIsBtn.setAttribute("style", "opacity: 0.4");
+    thisIsBtn.disabled = true;
+  }
```
```suggestion
  const isValid = inputs[0].value && inputs[1].value
  thisIsBtn.setAttribute("style", `opacity: ${isValid ? "1" : "0.4"}`);
  thisIsBtn.disabled = !isValid;
```
이런식으로 사용해줄 수도 있습니다.
1. 삼항연산자를 사용했습니다 (참고 : https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
2. 그 자체로 boolean 값을 갖는 성질을 이용할 수 있습니다

saengmotmi commented on students/11th/seunghaoh/js/main.js:
```diff
@@ -0,0 +1,46 @@
+
+  profileName.innerHTML = "profileName";
+  commentWritten.innerHTML = commentWrittenVal;
+  likeIcon.innerHTML = `<svg
```
템플릿 리터럴 + innerHTML = 👏👏👏

saengmotmi commented on students/11th/seunghaoh/js/main.js:
```diff
@@ -0,0 +1,46 @@
+  newComment.appendChild(likeIcon);
+}
+
+let commentWritten = document.getElementsByClassName("commentWritten")[0];
```
let으로 선언하신 이유가 특별히 있을까요?

saengmotmi commented on students/11th/seunghaoh/style/login.css:
```diff
@@ -0,0 +1,140 @@
+:root {
+  /* color */
+  --white-color: white;
+  --grey-light-color: #fafafa;
+  --grey-medium-color: #dbdbdb;
+  --grey-dark-color: #8e8e8e;
+  --blue-color: #0095f6;
```
변수 사용 굿굿 👍

saengmotmi commented on students/11th/seunghaoh/js/main.js:
```diff
@@ -0,0 +1,46 @@
+  let commentWrittenVal = document.getElementsByClassName("commentWritten")[0]
+    .value;
+  let likeIconElement = document.getElementsByClassName("likeIcon")[0];
+  console.log(likeIconElement);
```
테스트 끝난 console.log 삭제해주세요

joonsikyang commented:
> 자바스크립트 댓글 기능 구현에 어려움이 많았습니다. 어떤 순서로 코딩을 해야할 지 생각하는 것이 어렵습니다.
> 코드가 깔끔하지 않은 것 같습니다. 리뷰 부탁드립니다.

두 질문 모두 절대적인 시간 투자가 필요한 것 같습니다. 2주 진행된 상황에서 당연히 어려운 거라서 앞으로 리액트 진행하고 프로젝트 진행하면서 계속해서 좋아질거에요! 조급하지만 않게 꾸준히 해주시면 해결 될 문제입니다. 리뷰 반영해서 코드 최대한 깔끔하게 리팩토링 부탁드립니다.

dev-bomdong committed:
Add: login  페이지 기능 완성, main 페이지 진행중
- login form 레이아웃 작성 완료
- login email,password 유효성 검사 후 버튼 활성화 기능 구현
- main :마크업 완료

dev-bomdong committed:
[Mission 1~4완료] login 페이지 css수정, main 페이지 기능완성
- login 페이지 : class명 수정 위주 코드 리팩토링
- main 페이지 : css구현
- main 페이지 : 댓글 입력 후 엔터입력/버튼 클릭시 새로운 댓글 생성 기능 구현

dev-bomdong committed:
Refactor: 로그인 페이지, 메인 페이지 리팩토링
- login page : 중복코드 삭제
- main page : 중복코드 삭제, 불필요한 div삭제, class명 정리

dev-bomdong committed:
ADD: [Mission 5 완료]로그인페이지 ID, PW validation
- id : '@' 포함 / pw :  5글자 이상 일 때 로그인 버튼 활성화 기능 구현

dev-bomdong committed:
ADD: 메인페이지 댓글 삭제기능
- main page 버튼 클릭시 생성했던 댓글 삭제 기능 구현

dev-bomdong committed:
ADD: 메인페이지 댓글 좋아요/취소 기능 완료
- main page 하트 클릭시 붉은 색으로 변경
- main page 붉은 하트 클릭시 다시 빈 하트로 변경
- main page 댓글 삭제 버튼 변경
